File: app/src/main/java/com/pixels/blockies/app/game/figures/FigureL.java
package com.pixels.blockies.app.game.figures;

/**
 * Created by keinmark on 30.04.14.
 */
public class FigureL extends AbstractFigure {

    private int[][][] figures = new int[][][]{
            {
                    {-1, -1, 2},
                    {2, 2, 2}
            },
            {
                    {2, -1},
                    {2, -1},
                    {2, 2}
            },
            {
                    {2, 2, 2},
                    {2, -1, -1}
            },
            {
                    {2, 2},
                    {-1, 2},
                    {-1, 2}
            }
    };

    @Override
    protected int getFigureCount() {
        return figures.length;
    }

    @Override
    public int[][] get() {
        return figures[getCurrentRotation()];
    }

    @Override
    public int[][] getNext() {
        return figures[getNextRotation()];
    }
}

File: app/src/main/java/com/pixels/blockies/game/game/figures/FigureL.kt
package com.pixels.blockies.game.game.figures

import com.pixels.blockies.game.game.Grid

class FigureL : AbstractFigure() {
    private val figures = arrayOf(
        arrayOf<IntArray?>(
            intArrayOf(Grid.Companion.EMPTY, Grid.Companion.EMPTY, 2),
            intArrayOf(2, 2, 2)
        ),
        arrayOf<IntArray?>(
            intArrayOf(2, Grid.Companion.EMPTY),
            intArrayOf(2, Grid.Companion.EMPTY),
            intArrayOf(2, 2)
        ),
        arrayOf<IntArray?>(
            intArrayOf(2, 2, 2),
            intArrayOf(2, Grid.Companion.EMPTY, Grid.Companion.EMPTY)
        ),
        arrayOf<IntArray?>(
            intArrayOf(2, 2),
            intArrayOf(Grid.Companion.EMPTY, 2),
            intArrayOf(Grid.Companion.EMPTY, 2)
        )
    )

    override fun getFigureCount(): Int {
        return figures.size
    }

    override fun get(): Array<IntArray?> {
        return figures[currentRotation]
    }

    override fun getNext(): Array<IntArray?> {
        return figures[nextRotation]
    }
}